homeassistant/components/isy994/climate.py

```diff
@@ -0,0 +1,345 @@
+
+    @property
+    def operation_list(self):
+        """Return the list of available operation modes."""
+        return ISY_OPERATION_LIST
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self.value
+
+    @property
+    def target_temperature_step(self):
+        """Return the supported step of target temperature."""
+        return 1
+
+    @property
+    def target_temperature_high(self):
+        """Return the highbound target temperature we try to reach."""
+        return self._target_temp_high
+
+    @property
+    def target_temperature_low(self):
+        """Return the lowbound target temperature we try to reach."""
+        return self._target_temp_low
+
+    @property
+    def current_hold_mode(self):
+        """Return the current hold mode, e.g., home, away, temp."""
+        return self._current_hold_mode
+
+    @property
+    def is_on(self):
+        """Return if the unit is on or off based on operation status."""
+        if self._current_status not in [STATE_OFF, STATE_IDLE, None]:
+            return True
+        return False
+
+    @property
+    def fan_list(self):
+        """Return the list of available fan modes."""
+        return [STATE_AUTO, STATE_ON]
+
+    @property
+    def current_fan_mode(self) -> str:
+        """Return the current fan mode ie. auto, on."""
+        return self._fan_mode
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return ISY_SUPPORTED_MODES
+
+    @property
+    def min_temp(self):
+        """Return the minimum temperature."""
+        return convert_temperature(DEFAULT_MIN_TEMP, TEMP_CELSIUS,
+                                   self.temperature_unit)
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature."""
+        return convert_temperature(DEFAULT_MAX_TEMP, TEMP_CELSIUS,
+                                   self.temperature_unit)
+
+    @property
+    def device_state_attributes(self):
+        """Return the device specific state attributes."""
+        return None
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        if 'target_temp_high' in kwargs and kwargs['target_temp_high'] != \
+                self._target_temp_high:
+            self._node.climate_setpoint_cool(int(kwargs['target_temp_high']))
+            self._target_temp_high = kwargs['target_temp_high']
+        if 'target_temp_low' in kwargs and kwargs['target_temp_low'] != \
+                self._target_temp_low:
+            self._node.climate_setpoint_heat(int(kwargs['target_temp_low']))
+            self._target_temp_low = kwargs['target_temp_low']
+        self.schedule_update_ha_state()
+
+    def set_fan_mode(self, fan_mode):
+        """Set new target fan mode."""
+        _LOGGER.debug('Requested fan mode %s', fan_mode)
+        # pylint: disable=protected-access
+        self._node._fan_mode(fan_mode)
```

Comment:
Why do we need to access private methods? It would be best to either use the intended way of changing mode or to change the library to accessing a private method.